### DISCUSSION TITLE
light_gcn graph micro batching

### DISCUSSION THREAD

**Discussion by sorushsaghari** [pyg-team/pytorch_geometric#8721] [Q&A]
Hi everyone. torch_geometrics has a lightgcn model implemented which on the `forward` method has an edge_index param.
Because my dataset is large I can't load it into memory. is there a solution like `loader.DataLoader` to micro-batch `edge_index`. or this neural network need whole data at the same time and I can't do anything

**Comment by akihironitta** [comment]
Hi @sorushsaghari, you can utilise `NeighborLoader` (or `LinkNeighborLoader`) for your use-case. Have you had a look at the docs by any chance? https://pytorch-geometric.readthedocs.io/en/2.4.0/tutorial/neighbor_loader.html

**Reply by sorushsaghari** [reply]
it seems to be a good solution. so you mean i can sample `edge_index` for each node i want to train ? and reduce my `edge_index` size?

**Reply by akihironitta** [reply]
Yes.
